Added: kernbypassd

ichitaso opened the pull request:
Automatically enabled on ldrestart and Re-Jailbreak

level3tjg committed:
Notify from SpringBoard

level3tjg committed:
Fix other tweaks & Choicy compat

Commit:
Added: kernbypassd

Automatically enabled on ldrestart and Re-Jailbreak

Randomblock1 commented:
Can confirm this works
iPhone 6 12.4.5
iPhone 8+ 13.3

jedis00 commented:
Doesn’t work on iOS 12.4. Unc0v3r

[exec dpkg -i] ...
bash-5.0# dpkg -i "/var/mobile/Documents/jp.akusio.kernbypass_0.0.3_iphoneos-arm.deb" ;
Selecting previously unselected package jp.akusio.kernbypass.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 7317 files and directories currently installed.)
Preparing to unpack .../jp.akusio.kernbypass_0.0.3_iphoneos-arm.deb ...
Unpacking jp.akusio.kernbypass (0.0.3) ...
Setting up jp.akusio.kernbypass (0.0.3) ...
/var/MobileSoftwareUpdate/mnt1 FOLDER NOT FOUND
mkdir: cannot create directory '/var/MobileSoftwareUpdate/mnt1': Operation not permitted
chmod: cannot access '/var/MobileSoftwareUpdate/mnt1': No such file or directory
chown: cannot access '/var/MobileSoftwareUpdate/mnt1': No such file or directory
/var/MobileSoftwareUpdate/mnt1 FOLDER CREATED SUCCESS
/usr/bin/changerootfs &
host: 0xE07
I think you're using unc0ver, But load it anyway.
sec_cstring_addr: 0xFFFFFFF01FCA8DC4, sec_cstring_sz: 0x48DF6
sec_text_addr: 0xFFFFFFF01FD50000, sec_text_sz: 0x553DD8
allproc: 0xFFFFFFF02034B958
/var/lib/dpkg/info/jp.akusio.kernbypass.postinst: line 16:  8239 Segmentation fault: 11  /usr/bin/changerootfs
/var/lib/dpkg/info/jp.akusio.kernbypass.postinst: line 18: disown: %1: no such job
disown %1
RUNNING DAEMON
bash-5.0#

Randomblock1 commented:
You need to create /var/MobileSoftwareUpdate/mnt1.
For some reason, when you ran dpkg, you didn’t give it root access, somehow.

jedis00 commented:
Tried that. Root doesn’t have access. Neither does mobile. Installed sudo and added mobile to sudoers file. Sudo mkdir can create it but chrootfs still segmentation faults.

jedis00 commented:
iPhone:/var root# ls -lsha MobileSoftwareUpdate    
    total 8.0K                                         
    0 drwxr-xr-x  4 root wheel  128 May  8 01:03 .  
    0 drwxr-xr-x 39 root wheel 1.3K Apr 13 13:42 .. 
    4.0K -rw-r--r--  1 root wheel  505 May  3 03:11 last_update_result.plist                               
    4.0K -rw-r--r--  1 root wheel 3.4K May  3 03:11 restore.log                                            
    iPhone:/var root# whoami                           
    root                                               
    iPhone:/var root# mkdir -p /var/MobileSoftwareUpdate/mnt1                                               
    mkdir: cannot create directory ‘/var/MobileSoftwareUpdate/mnt1’: Operation not permitted

Randomblock1 commented:
I don't think your rootfs is mounted read/write, it might be read-only. Try rejailbreaking?
This isn't an issue with this PR btw, it's your jailbreak

jedis00 commented:
It’s not selected read only in unc0ver. But yes, it’s with the tweak itself and still exists with this PR. I opened an issue.

Randomblock1 commented:
No, it isn’t with this tweak.
The problem is related to Unc0ver and how you can’t create the mnt1 folder. It’s not the tweak’s fault. This PR works perfectly fine if mnt1 exists.
I don’t know why it won’t let you, I’d recommend asking for support on the Jailbreak subreddit/discord.

lhvy commented:
Can confirm this is working on iPhone XS 13.3

NyaMisty committed:
Support fakevar with dmg attach

NyaMisty committed:
Support devfs build

NyaMisty committed:
Merge conflict

NyaMisty committed:
Don't use -a switch

NyaMisty committed:
Fix support for u0

jailbreaker17 commented:
i noticed a bug after i installed 0.0.3 using unc0ver 5.0.1. im using ios 13.5 and iPhone 11 Pro Max and after i got the tweak installed then i proceeded to re-jailbreak with unc0ver while jailbroken it gets stuck with 17/31 Purging Software Updates. The only way to get out of this is to do a force reboot. 
![photo_2020-05-30_16-00-29](https://user-images.githubusercontent.com/65879720/83323056-d15fe580-a28e-11ea-95bb-0cf700968a96.jpg)

After force rebooting, I jailbreak as normal and I noticed that it also makes the process called com.apple.MobileSoftwareUpdate.CleanupPreparePathService CPU usage to 100% all the time after I rejailbreak after the force reboot. The only way to get out of this 100% CPU usage is to do a clean reboot and re-jailbreak then do an ldrestart.  

It will trigger the same cycle again if you re-jailbreak while jailbroken. 
![photo_2020-05-30_16-13-27](https://user-images.githubusercontent.com/65879720/83323332-a6769100-a290-11ea-9fb2-99da1ea38035.jpg)

lhvy commented:
Yeah I am having the same issue as well unfortunately, for now just uninstalling. Hopefully can be fixed soon.

jailbreaker17 commented:
Which of the two problems are you facing?

Randomblock1 commented:
This is due to U0 trying to remove everything in MobileSoftwareUpdate, however it cannot because, well, it would be pretty bad if it could delete /.
The fix: run ‘umount -f /var/MobileSoftwareUpdate/mnt1’ as root before rebooting

lhvy commented:
Apologies for not being clear enough, I am facing both the issue with unc0ver getting stuck and com.apple.MobileSoftwareUpdate.CleanupPreparePathService at 100% CPU frequently.

Randomblock1 commented:
My guess is the daemon is also trying to delete /var/MobileSoftwareUpdate/mnt1/. You can probably disable it with DaemonDisabler and be fine.

nguyenthienhoan commented:
> Automatically enabled on ldrestart and Re-Jailbreak

Can you switch from AppList to LibSparkAppList ? AppList make my device slowly and delay.

XsF1re committed:
Support for iOS 14

Support for iOS 14

XsF1re committed:
Add deb file

XsF1re committed:
Update README, Dependencies, and plist path

- Update README and tutorial
- Remove depends com.comex.attachanddetach.
- Change plist path for hide detection.

XsF1re committed:
Upload deb file

XsF1re committed:
Delete jp.akusio.kernbypass_0.0.3_iphoneos-arm.deb

XsF1re committed:
Fix preference not working on arm64e device.

Fix preference not working on arm64e device.

Commit:
Merge remote-tracking branch 'ichitaso/master' into unofficial

Commit:
Merge pull request #1 from ichitaso/unofficial

Unofficial

Commit:
unofficial version 0.0.4

Commit:
Merge pull request #2 from ichitaso/unofficial

unofficial version 0.0.4

Commit:
Fix: kernbypassd doesn't work

Commit:
Merge pull request #3 from ichitaso/unofficial

Fix: kernbypassd doesn't work

Commit:
Added Icon

Commit:
Merge pull request #4 from ichitaso/unofficial

Added Icon

XsF1re committed:
Support iOS/iPadOS 14.2

now support iOS/iPadOS 14.2 and updated libdimentio.

XsF1re committed:
Update README.md

XsF1re committed:
Possibly fix for Odyssey

Commit:
Merge remote-tracking branch 'ichitaso/master'

Commit:
v0.0.5

Added: CCSupport addon
changing: to see exactly running changerootfs works
changing: when you run kernbypassd from settings, it is called from SpringBoard (Enable only)

Runs on SpringBoard because the changerootfs will be killed when called from settings (Disabled after restart of SpringBoard)
